umpleonline/scripts/jjs/jjs_parse.js

```diff
@@ -0,0 +1,117 @@
+var JJsParse = {
+	addToArray: function (item) {
+		arr = new Array();
+		arr.push(item);
+		return arr;
+	},
+
+	addAttributes: function (attrs) {
+		// { "type" : "String", "name" : "startTime", "modifier" : "", "traceColor" : "black" }
+		var i, text, attribute, max = attrs.length, attributes = new Array();
+		for (i = 0; i < max; i++) {
+			attribute = attrs[i];
+			text = this.makeModifier(attribute.modifier) + " " + attribute.name + ": " + attribute.type;
+			attributes.push(text);
+		}
+		return attributes;
+	},
+
+	addMethods: function(meths) {
+		// {type: "void", name: "setDoorTimer", parameters: "Integer", visibility: "public", isAbstract: "false"}
+		// console.log(meths);
+		var i, text, method, max = meths.length, methods = new Array();
+		for (i = 0; i < max; i++) {
+			method = meths[i];
+			text =  this.makeModifier(method.visibility) + " " + method.name + "(" + method.parameters + ") : " + method.type;
+			methods.push(text);  
+		}
+		return methods;
+	},
+
+	makeModifier: function(modifier) {
+		var symbol = "";
+		switch (modifier) {
+			case "public":
+				symbol = "+";
+				break;
+			case "package":
+				symbol = "~";
+				break;
+			case "protected":
+				symbol = "#";
+				break;
+			case "private":
+				symbol = "-";
+				break;
+			default:
+				symbol = "-";
+			}
+		return symbol;
+	},
+
+	makeClasses: function (model) {
+		var i, text, UMLclass, max = model.umpleClasses.length, classes = new Array();
+		var new_class;
+
+		for (i = 0; i < max; i++) {
```

Comment:
Ah! Nice pattern, it is new to me.
It's much cleaner now, and I will implement this in the other methods that also apply.